coffeecard/CoffeeCard.WebApi/Helpers/PageUtils.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Threading.Tasks;
+using CoffeeCard.Common.Errors;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Serilog;
+
+
+namespace CoffeeCard.WebApi.Helpers
+{
+    /// <summary>
+    /// Convenience methods to be used on the user-facing webpages
+    /// </summary>
+    public static class PageUtils
+    {
+        public static async Task<IActionResult> SafeExecuteFunc(Func<Task<IActionResult>> func, PageModel model)
+        {
+            try
+            {
+                return await Task.Run(func);
+            }
+            catch (ApiException ex)
+            {
+                // Not logging this, since we should have logged the cause when we created it.
+                if (ex.StatusCode == StatusCodes.Status401Unauthorized)
+                {
+                    setMessage("Error", "Looks like the link you used has expired or already been used. Request a new link and try again", model);
```

Comment:
It is not clear to me in what case we throw a 401 for an already-used token?

Reply:
Currently, we do not, though we might in the future if we make the email confirmation and delete account methods, check if the token has been used

Reply:
Mjah :(